### PR TITLE
Feature Policy: fix tests that time out

### PR DESCRIPTION
**Pull request by zcorpan**
Part of #11269.

### CODE REVIEW

File: feature-policy/resources/picture-in-picture.js [hunk 1]
```diff
@@ -3,14 +3,18 @@ function isPictureInPictureAllowed() {
...
+      try {
+        video.requestPictureInPicture()
+        .then(() => resolve(document.pictureInPictureEnabled))
+        .catch(e => {
+          if (e.name == 'NotAllowedError')
+            resolve(document.pictureInPictureEnabled);
+          else
+            resolve(false);
+        });
+      } catch(e) {
```
Comment (Hexcles) [inline feature-policy/resources/picture-in-picture.js]: I don't know much about feature policy or picture in picture, but this broad exception catch smells a bit. Are you trying to catch exceptions from `video.requestPictureInPicture` (e.g. undefined)?

**Review by Hexcles — APPROVED** [review]
This generally LGTM, but I'd like confirmation from @clelland .

File: feature-policy/resources/picture-in-picture.js (hunk 1 above)
Comment (zcorpan) [reply feature-policy/resources/picture-in-picture.js]: Yeah. I suppose I can do `t.step_func` instead.

Comment (zcorpan) [reply feature-policy/resources/picture-in-picture.js]: The function is sometimes called when there's no Test object. But I've changed to an `if` check instead.

**Comment by beaufortfrancois** [comment]
Thanks for this patch.
Let me have a look.

**Comment by beaufortfrancois** [comment]
I've uploaded a patch at https://chromium-review.googlesource.com/c/chromium/src/+/1107626 that fixes the issue. It will be upstreamed soon.

**Comment by Hexcles** [comment]
We can either land that CL and rebase-merge this PR, or land this PR and abandon that CL. Either is fine to me.

**Comment by beaufortfrancois** [comment]
I think https://chromium-review.googlesource.com/c/chromium/src/+/1107626 addresses all time out issues. Please let me know why you're changing some `async_test` to `promise_test`.

**Comment by zcorpan** [comment]
I figured that since the tests were using promises they ought to use promise_test, but maybe it works OK either way.

**Comment by beaufortfrancois** [comment]
I'm not sure to understand why but it looks like one test times out in https://travis-ci.org/web-platform-tests/wpt/jobs/396384904 while it doesn't for me locally. Do you folks have any idea why this may happening?

```js

promise_test(async t => {
  const video1 = await loadVideo();
  const video2 = await loadVideo();
  return callWithTrustedClick(() => {
    const first = video1.requestPictureInPicture();
    const second = video2.requestPictureInPicture();
    return Promise.all([
      first,
      promise_rejects(t, 'NotAllowedError', second)
    ]);
  });
}, 'request Picture-in-Picture consumes user gesture');
```

and we have now both promises which always reject... 

```js
if (!('pictureInPictureEnabled' in document)) {
  HTMLVideoElement.prototype.requestPictureInPicture = function() {
    return Promise.reject('Picture-in-Picture API is not available');
  }
}
```

**Comment by Hexcles** [comment]
#11589 has landed and addressed all timeout on both Firefox and Chrome. This is no longer necessary, but feel free to reopen if I missed anything.

**Pull request closed without merging**